subsys/usb/usb_c/usbc_pe_common.c

```diff
@@ -229,6 +234,40 @@ void pe_message_sent(const struct device *dev)
 	atomic_set_bit(pe->flags, PE_FLAGS_TX_COMPLETE);
 }
 
+static bool pe_soft_reset_is_required(const struct device *dev, const enum pd_packet_type type)
+{
+	struct usbc_port_data *data = dev->data;
+	struct policy_engine *pe = data->pe;
+
+	/* Protocol Error not on SOP */
+	if (type != PD_PACKET_SOP) {
+		return false;
+	}
+
+	/* Protocol error on SOP during non-AMS */
+	if (atomic_test_bit(pe->flags, PE_FLAGS_INTERRUPTIBLE_AMS)) {
+		return true;
+	}
+
+	/* Not in an explicit contract */
+	if (!atomic_test_bit(pe->flags, PE_FLAGS_EXPLICIT_CONTRACT)) {
+		/* Protocol Errors occurred on SOP during an Interruptible AMS */
+		if (!atomic_test_bit(pe->flags, PE_FLAGS_INTERRUPTIBLE_AMS)) {
```

Comment:
This will always be true because of the check on line 248, right?

Reply:
I had the logic all wrong